openapi/data_repository_service.swagger.yaml

```diff
@@ -192,144 +152,156 @@ definitions:
-      size:
+        description: |-
+          A string that can be used to name an `Object`.
+      type:
         type: string
-        format: int64
-        description: >-
-          The cumulative size, in bytes, of all Data Objects and Bundles listed in
-          the `contents` field.
+        enum:
+        - blob
+        - bundle
+      blob:
+        $ref: '#/definitions/BlobDescription'
+        description: |-
+          If `type` is blob, then this property must be present and the `bundle` property must be absent or null
+      bundle:
+        $ref: '#/definitions/BundleDescription'
+        description: |-
+          If `type` is bundle, then this property must be present and the `blob` property must be absent or null
+  CommonDescription:
+    type: object
+    required: ['created']
+    properties:
       created:
         type: string
         format: date-time
-        description: >-
-          Timestamp of Bundle creation in RFC3339.
+        description: |-
+          Timestamp of object creation in RFC3339.
       updated:
         type: string
         format: date-time
         description: >-
-          Timestamp of Bundle update in RFC3339, identical to create timestamp in
-          systems that do not support updates.
+          Timestamp of `Object` update in RFC3339, identical to create timestamp in systems
+          that do not support updates.
       version:
         type: string
         description: >-
-          A string representing a version. (Some systems may use checksum, a
-          RFC3339 timestamp, or an incrementing version number.)
-      checksums:
-        type: array
-        description: |-
-          The checksum of the Data Bundle. At least one checksum must be provided.
-          
-          The Data Bundle checksum is computed over a sorted concatenation of all 
-          the checksums (names not included) within the top-level 'contents' of the
-          Bundle (not recursive). The list of Data Object or Bundle checksums are 
-          sorted alphabetically (hex-code) before concatenation and a further checksum
-          is performed on the concatenated checksum value.
-          Example below:
-          Data Ojects:
-            md5(DO1) = 72794b6d30bc86d92e40a1aa65c880b8
-            md5(DO2) = 5e089d29a18954e68a78ee6a3c6edabd
-          Data Bundle:
-          DB1 = md5( concat( sort( md5(DO1), md5(DO2) ) ) )
-              = md5( concat( sort( 72794b6d30bc86d92e40a1aa65c880b8, 5e089d29a18954e68a78ee6a3c6edabd ) ) )
-              = md5( concat( 5e089d29a18954e68a78ee6a3c6edabd, 72794b6d30bc86d92e40a1aa65c880b8 ) )
-              = md5( 5e089d29a18954e68a78ee6a3c6edabd72794b6d30bc86d92e40a1aa65c880b8 )
-              = f7a29a0422e7d870b10839ad6c985079
-        items:
-          $ref: '#/definitions/Checksum'
+          A string representing a version.
+
+          (Some systems may use checksum, a RFC3339 timestamp, or an incrementing version number.)
       description:
```

Comment:
I might retract (or revise) my comment above regarding keywords... I have no objections to a `description` property that also has a Swagger/OpenAPI/JSON Schema `description`.